skydriver/images.py

```diff
@@ -0,0 +1,111 @@
+"""Utilities for dealing with docker/cvmfs/singularity images."""
+
+import re
+from pathlib import Path
+
+import cachetools.func
+import requests
+
+from .config import LOGGER
+
+# ---------------------------------------------------------------------------------------
+# constants
+
+
+_IMAGE = "skymap_scanner"
+_SKYSCAN_DOCKER_IMAGE_NO_TAG = f"icecube/{_IMAGE}"
+
+DOCKERHUB_API_URL = (
+    f"https://hub.docker.com/v2/repositories/{_SKYSCAN_DOCKER_IMAGE_NO_TAG}/tags"
+)
+
+# cvmfs singularity
+_SKYSCAN_CVMFS_SINGULARITY_IMAGES_DPATH = Path(
+    "/cvmfs/icecube.opensciencegrid.org/containers/realtime/"
+)
+VERSION_REGEX_MAJMINPATCH = re.compile(r"\d+\.\d+\.\d+")
+VERSION_REGEX_PREFIX_V = re.compile(r"v\d+(\.\d+(\.\d+)?)?")
+VERSION_REGEX_MAJ_OR_MAJMIN = re.compile(r"\d+(\.\d+)?")
+
+# clientmanager
+CLIENTMANAGER_IMAGE_WITH_TAG = "ghcr.io/wipacrepo/skydriver:latest"
```

Comment:
This should not be `latest`, and should in fact probably be configured in the yaml somewhere.  (Isn't it the same image that the rest api is running?)

Reply:
I figured this is fine since skydriver will always be "latest" via the argocd bot's updating (minus some delay between syncing)

Reply:
> Isn't it the same image that the rest api is running?

Reply:
Technically yes, but they could be slightly out of sync. And we'll also have to pull another tag (even if it should mostly be cached layers).  I also dislike that it is hard-coded, just in case we move where we store our images.

Reply:
Okay, I see that. Is there a way to have the argocd bot update an env var with that tag (or bind it to the current tag)? https://github.com/WIPACrepo/kubernetes/blob/master/ddn-k8s-2/argocd/skydriver/templates/rest-Deployment.yaml#L23

Reply:
Yes, you should be able to specify an env variable with the value of `"{{ .Values.image.repository }}:{{ .Values.image.tag }}"`.